pdf/pdfrenderthread.cpp

```diff
@@ -279,19 +279,21 @@ void PDFRenderThread::cancelRenderJob(int index)
 void PDFRenderThread::prioritizeJob(int index, int size)
 {
     QMutexLocker locker(&d->thread->mutex);
-    for (QList<PDFJob *>::iterator it = d->thread->jobQueue->begin(); it != d->thread->jobQueue->end(); ++it) {
+    RenderPageJob *last = nullptr;
+    for (QList<PDFJob *>::iterator it = d->thread->jobQueue->begin(); it != d->thread->jobQueue->end();) {
         PDFJob *j = *it;
         if (j->type() == PDFJob::RenderPageJob) {
             RenderPageJob *rj = static_cast<RenderPageJob *>(j);
             if (rj->m_index == index && rj->renderWidth() == size) {
-                if (it != d->thread->jobQueue->begin()) { // If it is already at the front, just abort..
-                    d->thread->jobQueue->erase(it);
-                    d->thread->jobQueue->push_front(j);
-                }
-                return;
+                it = d->thread->jobQueue->erase(it);
+                last = rj;
```

Comment:
Ok, I've finally kept the prioritize() function as it is, with an additional parameter subpart, that allow to update and prioritize the request at the same time.